index.js

```diff
@@ -326,6 +326,7 @@
    */
 
   Assertion.prototype.match = function (regexp) {
+    if (!regexp.exec) regexp = new RegExp(regexp);
```

Comment:
Doesn't `new RegExp` call toString implicitly and I would argue the new
String cases are a) uncommon b) not as important as cases such as {exec: 1}

On Mon, Apr 20, 2015 at 10:17 PM, Jason Karns <notifications@github.com>
wrote:

> In index.js
> <https://github.com/Automattic/expect.js/pull/132#discussion_r28745609>:
>
> > @@ -326,6 +326,7 @@
> >     */
> >
> >    Assertion.prototype.match = function (regexp) {
> > +    if (!regexp.exec) regexp = new RegExp(regexp);
>
> i disagree:
>
> typeof new String("foo")
> > "object"
>
> Further, as proper duck-typists, we don't actually care that the type is a
> string. Only that the RegExp constructor accepts it. Of course, at the
> moment, that only applies to strings and other regexes. But as a design
> decision, we shouldn't be adding that arbitrary limitation.
>
> Further, we don't actually care that regexp is a regular expression. Only
> that we can invoke exec. A consumer is more than capable of passing in a
> custom object that has an exec method that does something appropriate to
> their custom class.
>
> Technically, I would consider it even more bulletproof if we did: if
> (!regexp.exec) regexp = new RegExp(regexp.toString()); This way we ensure
> that RegExp is given a string, (as is it's current contract) but also
> allows consumers to provide custom objects that have an appropriate
> toString implementation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/expect.js/pull/132/files#r28745609>.
>